src/librbd/object_map/DiffRequest.cc

```diff
@@ -233,15 +233,12 @@ void DiffRequest<I>::handle_load_object_map(int r) {
   }
   ldout(cct, 20) << "computed overlap diffs" << dendl;
 
-  bool diff_from_start = (m_snap_id_start == 0);
   auto end_it = m_object_map.end();
   for (; it != end_it; ++it, ++diff_it, ++i) {
     uint8_t object_map_state = *it;
     if (object_map_state == OBJECT_NONEXISTENT) {
       *diff_it = DIFF_STATE_HOLE;
-    } else if (diff_from_start ||
-               (m_object_diff_state_valid &&
-                object_map_state != OBJECT_EXISTS_CLEAN)) {
+    } else if (m_current_snap_id != m_snap_id_start) {
       // diffing against the beginning of time or image was grown
```

Comment:
> Is this comment still valid? And also "diffing against a snapshot" comment in "else" close below.

Yes, both are still valid.  `m_current_snap_id != m_snap_id_start` is always true when diffing against the beginning of time because `m_snap_id_start` is 0 in that case.  The only time `m_current_snap_id != m_snap_id_start` is false is when the diff is against a snapshot and we are looking at that snapshot's (i.e. starting snapshot) object map.

Reply:
makes sense. Thanks.